core/src/main/scala/kafka/server/DynamicConfigManager.scala

```diff
@@ -138,7 +138,10 @@ class DynamicConfigManager(private val zkClient: ZkClient,
             case Some(value: String) => value
             case _ => throw new IllegalArgumentException("Config change notification does not specify 'entity_name'. Received: " + json)
           }
-          configHandler(entityType).processConfigChanges(entity, AdminUtils.fetchEntityConfig(zkClient, entityType, entity))
+
+          val entityConfig = AdminUtils.fetchEntityConfig(zkClient, entityType, entity)
+          logger.info(s"Processing override for entityType: $entityType, entity: $entity with config: $entityConfig")
+          configHandler(entityType).processConfigChanges(entity, entityConfig)
```

Comment:
Good catch! It seems like we had this issue in TopicConfigManager as well. The reason it works for  topic configs currently is because in KafkaServer we do a AdminUtils.fetchAllTopicConfigs. This way all configs are explicitly loaded prior to receiving config changes. I think we have 2 options in this patch:

1. Keep the existing code. Read all entity configs at startup for ClientIds similar to topic configs.
2. Reload all existing entityPath for topics and clients. With this approach, we will need to change LogManager to not read all topic configs at startup or they will get applied twice (which should be harmless but unnecessary).

1 is safer but needs to be cleaned up later. 2 is probably the right thing but not sure if we want to do this right before the 0.9 branch is cut. 

Additionally, I'll look at KAFKA-2547 but my initial reaction is to do it in a followup patch. My only concern is that this change can get a little tricky and I want to be careful about changes right before 0.9. Let me know what you think. I'm not opposed to doing it in this patch if you think that is the right thing to do.